reconcile/dynatrace_token_provider/integration.py

```diff
@@ -69,7 +71,30 @@ def run(self, dry_run: bool) -> None:
         dependencies.populate()
         self.reconcile(dry_run=dry_run, dependencies=dependencies)
 
+    def _token_cnt(self, dt_tenant_id: str, ocm_env_name: str) -> None:
+        with self._lock:
+            if ocm_env_name not in self._managed_tokens_cnt:
+                self._managed_tokens_cnt[ocm_env_name] = {}
+            if dt_tenant_id not in self._managed_tokens_cnt[ocm_env_name]:
+                self._managed_tokens_cnt[ocm_env_name][dt_tenant_id] = 0
+            self._managed_tokens_cnt[ocm_env_name][dt_tenant_id] += 1
+
+    def _expose_token_metrics(self) -> None:
+        for ocm_env_name, counter_by_tenant_id in self._managed_tokens_cnt.items():
+            for dt_tenant_id, cnt in counter_by_tenant_id.items():
+                metrics.set_gauge(
+                    DTPTokensManagedGauge(
+                        integration=self.name,
+                        ocm_env=ocm_env_name,
+                        dt_tenant_id=dt_tenant_id,
+                    ),
+                    cnt,
+                )
+
     def reconcile(self, dry_run: bool, dependencies: Dependencies) -> None:
+        self._lock = Lock()
```

Comment:
why use lock, which part is run in threads?

Reply:
Currently single threaded, but I want to keep this thread-safe as it is quite possible that we might need concurrency.

Reply:
ok, then please move instance vars init (`_lock` and `_managed_tokens_cnt`) to `__init__` so we can clearly see what vars are available.

Reply:
Done :+1: